llama_index/schema.py

```diff
@@ -44,63 +92,119 @@ class BaseDocument(DataClassJsonMixin):
     This must be a flat dictionary, 
     and only uses str keys, and (str, int, float) values.
     """
-    extra_info: Optional[Dict[str, Any]] = None
-
-    def __post_init__(self) -> None:
-        """Post init."""
-        # assign doc_id if not set
-        if self.doc_id is None:
-            self.doc_id = get_new_id(set())
-        if self.doc_hash is None:
-            self.doc_hash = self._generate_doc_hash()
-
-        if self.extra_info is not None:
-            _validate_is_flat_dict(self.extra_info)
-
-    def _generate_doc_hash(self) -> str:
-        """Generate a hash to represent the document."""
-        doc_identity = str(self.text) + str(self.extra_info)
-        return sha256(doc_identity.encode("utf-8", "surrogatepass")).hexdigest()
+    metadata: Dict[str, Any] = Field(
+        default_factory=dict, description="A flat dictionary of metadata fields"
+    )
+    metadata_keys_to_exclude_for_embed: List[str] = Field(
```

Comment:
Include or exclude?

Reply:
Ah these metadata are excluded from metadata string. Maybe just call it embed_metadata or something similar?

Reply:
Yea mb, the description is old. I found excluding was easier than including (since the old behavior was to include everything)

Reply:
Basically it's a list of keys that are removed from the Metadata, controllable for embeddings and LLM calls

Reply:
changed the names to `excluded_embed_metadata_keys` and `excluded_llm_metadata_keys` -- a bit less verbose